R/checkDeps.R

```diff
@@ -26,18 +27,32 @@ checkDeps <- function(descriptionFile = ".",
                       dependencyTypes = c("Depends", "Imports", "LinkingTo"),
                       action = "stop") {
   stopifnot(all(dependencyTypes %in% c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")))
-  stopifnot(action %in% c("stop", "warn", "pass"))
+  stopifnot(action %in% c("stop", "warn", "pass", "ask"))
   allDeps <- desc::desc_get_deps(descriptionFile)
   deps <- allDeps[allDeps$type %in% dependencyTypes, ]
   requirementMet <- Map(checkRequirement, package = deps$package, version = deps$version)
 
-  if (!all("TRUE" == requirementMet)) {
-    message <- paste(requirementMet[which("TRUE" != requirementMet)],
-                     collapse = "\n  ")
-    if ("stop" == action) {
-      stop(message)
-    } else if ("warn" == action) {
-      warning(message)
+  if (!all(requirementMet == "TRUE")) {
+    missing <- requirementMet[requirementMet != "TRUE"]
+    msg <- paste(missing, collapse = "\n  ")
+    if (action == "ask" && any(grepl(": ", missing))) {
+      if (!requireNamespace("renv", quietly = TRUE) || is.null(renv::project())) {
+        stop("'action = ask' is only supported in an activated renv.")
+      }
+      message("missing requirements:\n  ", msg, "\n",
+              "Try to fix automatically? (y/n) ", appendLF = FALSE)
+      if (tolower(getLine()) %in% c("y", "yes")) {
+        renvInstall <- sub("^.+: ([^:]+)$", "\\1", grep(": ", missing, value = TRUE))
+        renv::install(renvInstall, prompt = FALSE)
+        renv::snapshot(prompt = FALSE)
+        # after installing, check again in new R session to avoid using already loaded packages
+        return(callr::r(function(...) lucode2::checkDeps(...),
```

Comment:
```suggestion
        return(callr::r(lucode2::checkDeps,
```
why do you have to wrap a function in a lambda function? Does it work if you don't?

Reply:
The unwrapped version does not ensure the lucode2 namespace is available/loaded, so it does not work. It's like copy pasting the checkDeps code here - the problem is helper functions defined in lucode2 like getLine cannot be found.